soundsgood.dsp

```diff
@@ -337,36 +337,83 @@ mscomp8i(target) =
-  crossover =
-    (
-      (crossoverFreqs<:par(i, Nr_crossoverFreqs, _<:si.bus(N)))
-     ,si.bus(N)
-    )
-    <: ro.interleave(N,Nr_bands)
-    :  par(i, N, fi.crossover8LR4)
-    :  ro.interleave(Nr_bands,N);
-  compressors =
-    (strength_array , thresh_array , att_array , rel_array , knee_array , link_array, ro.interleave(N,Nr_bands))
+
+  /* re-order, interleave output:  left gain [Nr_bands .. 1], left-signal , right gain [Nr_bands .. 1], right-signal
+   * Note: shelfcascade takes gains in reverse order */
+  groupGainOut  =
+    (ro.interleave(Nr_bands,N), si.bus(N))
+    : (ro.interleave(N, Nr_bands+1))
+    : par(i, N, ro.cross(Nr_bands),_);
+
+  /* ******* 8< *******/
+  // TODO: use co.peak_compression_gain_N_chan_db when it arrives in the current faust version
+  peak_compression_gain_mono_db(strength,thresh,att,rel,knee,prePost) =
+    abs : ba.bypass1(prePost,si.onePoleSwitching(att,rel)) : ba.linear2db : gain_computer(strength,thresh,knee) : ba.bypass1((prePost !=1),si.onePoleSwitching(rel,att))
+  with {
+    gain_computer(strength,thresh,knee,level) =
+      select3((level>(thresh-(knee/2)))+(level>(thresh+(knee/2))),
+        0,
+        ((level-thresh+(knee/2)) : pow(2)/(2*max(ma.EPSILON,knee))),
+        (level-thresh))
+      : max(0)*-strength;
+  };
+
+  peak_compression_gain_N_chan_db(strength,thresh,att,rel,knee,prePost,link,1) =
+    peak_compression_gain_mono_db(strength,thresh,att,rel,knee,prePost);
+
+  peak_compression_gain_N_chan_db(strength,thresh,att,rel,knee,prePost,link,N) =
+    par(i, N, peak_compression_gain_mono_db(strength,thresh,att,rel,knee,prePost))
+    <: (si.bus(N),(ba.parallelMin(N) <: si.bus(N))) : ro.interleave(N,2) : par(i,N,(it.interpolate_linear(link)));
+
+  /* ******* >8 *******/
+
+  compressor(N,prePost,strength,thresh,att,rel,knee,link) = peak_compression_gain_N_chan_db (strength,thresh,att,rel,knee,prePost,link,N);
+
+  comps(N) = (strength_array, thresh_array, att_array, rel_array, knee_array, link_array, ro.interleave (N,Nr_bands))
     : ro.interleave(Nr_bands,6+N)
-    : par(i, Nr_bands, compressor(meter(i+1),N,prePost)) ;
-  mixer = si.bus(N*Nr_bands):>si.bus(N);
+    : par(i, Nr_bands, compressor(N,prePost)) // : si.bus (N * Nr_bands)
+    : par(c, N, par(b, Nr_bands, meter(c+1, b+1)));
+
+  calcredux(N,freqs) = par(i, N, an.analyzer (6, freqs): ro.cross (Nr_bands)) : comps (N);
+
   outputGain = par(i, N, _*mscomp8i_outGain);
 
-  compressor(meter,N,prePost,strength,thresh,att,rel,knee,link) =
-    co.FFcompressor_N_chan(strength,thresh,att,rel,knee,prePost,link,meter,N);
+  /* TODO: separate %c%b in symbol name so that it is a valid C/C++ variable-name (ideally an underscore) %c_%b
+   * meanwhile this is safe since there are only 8 bands (1..9) and 2 channels.
+   */
+  meter(c,b) = _<: attach(_, (max(-40):min(0):vbargraph("v:soundsgood/t:expert/h:[6]mscomp_meter/[%c.%b][unit:dB][tooltip: gain reduction in db][symbol:msredux%{c}_%{b}]mscomp redux chn %c band %b", -3, 0)));
```

Comment:
This then becomes:
```faust
  meter(c,b) = _<: attach(_, (max(-3):min(0):vbargraph("v:soundsgood/t:expert/h:[6]mscomp_meter/[%b.%c][unit:dB][tooltip: gain reduction in db][symbol:msredux%b%c]mscomp redux band %b chn %c", -3, 0)));
```

Reply:
Done. Now the order is the same as before.
It'd be kinda cool to also include the actual freq-band range in the label or tooltip.

Reply:
[This commit](https://github.com/trummerschlunk/soundsgood/commit/e93c2e74ae2ccc2336d8fe22966e955ddb39d2c8) gives this routing:

![image](https://user-images.githubusercontent.com/7645711/185983251-2a6a56a7-f803-4f1f-8fb3-c7d105aaa157.png)

Each compressor block is an N channel compressor, where N =2 in this case.
The labels do not mach the actual channels.

Reply:
@x42 What does [your latest commit](https://github.com/trummerschlunk/soundsgood/compare/e93c2e74ae2ccc2336d8fe22966e955ddb39d2c8..67e0b526f616f4e1a3d731215c8f2f77f7d996e9) do?
So far I've been only testing the faust code, as I haven't gotten around to making a nix build for the GUI yet.

Reply:
NM, it's putting back the bypass in the brickwall, isn't it?
It looked like it was changing the meters, but that's just because a parameter got added, right?

Reply:
Yes rebased to master - I'm testing with `faust2lv2` and the parameters are matched to master branch.
What is the issue with compressor meter ordering?